Metrics check for process' network transmit/receive queue lengths

sivakom opened the pull request:
This sensu check returns the network transmit and receive queue length metrics for a specified process. Specifically, the script looks at all the sockets owned by the process, uses the `netstat -tpane` command and extracts the queue lengths for each socket from this command output.

Also fixed a minor bug that where the metrics weren't being summed up for all the pids.

Commit:
metrics check for process' network transmit/receive queue lengths

added criticality

remove criticality for network queue metrics

omgjlk commented on sensu/plugins/metrics-network-queues.py:
```diff
@@ -0,0 +1,112 @@
+
+    try:
+        process_sockets = subprocess.check_output \
+                          (('grep', '%d' % pid), stdin=netstat.stdout)
```
Uh, Python has better methods to find a substring within a string than shelling out to grep.

I'd recommend getting the output of netstat into a variable, and then using the native python tooling to inspect that output.

omgjlk commented:
This new module will need a criticality added, like we added in 4bf3b0650054077d6335fa923351892b5701b81c

sivakom commented:
Apologies, forgot about the python functionality that could substitute for grep. Also added criticality for both of these scripts. In addition, simplified and adapted the scripts to match style/syntax of other plugins in this repo

sivakom commented:
Sorry to bother you reviewers, but I'd greatly appreciate it if y'all could take a peek at this again. Thanks for your time.

omgjlk commented:
I think this is fine, @paulczar ?

paulczar commented on sensu/plugins/metrics-network-queues.py:
```diff
@@ -0,0 +1,116 @@
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+CRITICALITY = 'critical'
```
this is only needed for checks ... metrics do not need this as they do not create alerts.

paulczar commented:
@j2sol FYI criticality is only required for alerts.   metrics do not need.
